### ISSUE TITLE
ValueError when parsing /etc/os-release with unquoted URL value

### ISSUE THREAD

**Issue by zangafan** [cockpit-project/cockpit#19095]
### Explain what happens

I can't log in to Cockpit on my server running EndeavourOS. It just says wrong username or password on the webpage but the logs mention some sort of ValueError.

1. Navigate to localhost:9090
2. Log in with username and password
3. Says "wrong username or password"

### Version of Cockpit

295-1

### Where is the problem in Cockpit?

Unknown or not applicable

### Server operating system

Arch Linux

### What browsers are you using?

Firefox, Chrome

### System log

```shell
Jul 11 22:23:20 matt-server cockpit-ws[2663]: Traceback (most recent call last):
Jul 11 22:23:20 matt-server cockpit-ws[2663]:   File "/usr/bin/cockpit-bridge", line 8, in <module>
Jul 11 22:23:20 matt-server cockpit-ws[2663]:     sys.exit(main())
Jul 11 22:23:20 matt-server cockpit-ws[2663]:              ^^^^^^
Jul 11 22:23:20 matt-server cockpit-ws[2663]:   File "/usr/lib/python3.11/site-packages/cockpit/bridge.py", line 275, in main
Jul 11 22:23:20 matt-server cockpit-ws[2663]:     run_async(run(args), debug=args.debug)
Jul 11 22:23:20 matt-server cockpit-ws[2663]:   File "/usr/lib/python3.11/site-packages/cockpit/_vendor/systemd_ctypes/event.py", >
Jul 11 22:23:20 matt-server cockpit-ws[2663]:     asyncio.run(main, debug=debug)
Jul 11 22:23:20 matt-server cockpit-ws[2663]:   File "/usr/lib/python3.11/asyncio/runners.py", line 190, in run
Jul 11 22:23:20 matt-server cockpit-ws[2663]:     return runner.run(main)
Jul 11 22:23:20 matt-server cockpit-ws[2663]:            ^^^^^^^^^^^^^^^^
Jul 11 22:23:20 matt-server cockpit-ws[2663]:   File "/usr/lib/python3.11/asyncio/runners.py", line 118, in run
Jul 11 22:23:20 matt-server cockpit-ws[2663]:     return self._loop.run_until_complete(task)
Jul 11 22:23:20 matt-server cockpit-ws[2663]:            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
Jul 11 22:23:20 matt-server cockpit-ws[2663]:   File "/usr/lib/python3.11/asyncio/base_events.py", line 653, in run_until_complete
Jul 11 22:23:20 matt-server cockpit-ws[2663]:     return future.result()
Jul 11 22:23:20 matt-server cockpit-ws[2663]:            ^^^^^^^^^^^^^^^
Jul 11 22:23:20 matt-server cockpit-ws[2663]:   File "/usr/lib/python3.11/site-packages/cockpit/bridge.py", line 161, in run
Jul 11 22:23:20 matt-server cockpit-ws[2663]:     StdioTransport(asyncio.get_running_loop(), router)
Jul 11 22:23:20 matt-server cockpit-ws[2663]:   File "/usr/lib/python3.11/site-packages/cockpit/transports.py", line 459, in __ini>
Jul 11 22:23:20 matt-server cockpit-ws[2663]:     super().__init__(loop, protocol, stdin, stdout)
Jul 11 22:23:20 matt-server cockpit-ws[2663]:   File "/usr/lib/python3.11/site-packages/cockpit/transports.py", line 88, in __init>
Jul 11 22:23:20 matt-server cockpit-ws[2663]:     self._protocol.connection_made(self)
Jul 11 22:23:20 matt-server cockpit-ws[2663]:   File "/usr/lib/python3.11/site-packages/cockpit/protocol.py", line 148, in connect>
Jul 11 22:23:20 matt-server cockpit-ws[2663]:     self.do_ready()
Jul 11 22:23:20 matt-server cockpit-ws[2663]:   File "/usr/lib/python3.11/site-packages/cockpit/protocol.py", line 262, in do_ready
Jul 11 22:23:20 matt-server cockpit-ws[2663]:     self.do_send_init()
Jul 11 22:23:20 matt-server cockpit-ws[2663]:   File "/usr/lib/python3.11/site-packages/cockpit/bridge.py", line 133, in do_send_i>
Jul 11 22:23:20 matt-server cockpit-ws[2663]:     os_release=self.get_os_release(),
Jul 11 22:23:20 matt-server cockpit-ws[2663]:                ^^^^^^^^^^^^^^^^^^^^^
Jul 11 22:23:20 matt-server cockpit-ws[2663]:   File "/usr/lib/python3.11/site-packages/cockpit/bridge.py", line 116, in get_os_re>
Jul 11 22:23:20 matt-server cockpit-ws[2663]:     return dict(token.split('=', 1) for token in lexer)
Jul 11 22:23:20 matt-server cockpit-ws[2663]:            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
Jul 11 22:23:20 matt-server cockpit-ws[2663]: ValueError: dictionary update sequence element #11 has length 1; 2 is required
Jul 11 22:23:20 matt-server cockpit-session[2639]: pam_unix(cockpit:session): session closed for user root
```

**Comment by LionUCS** [comment]
Exact same problem here. Same OS, same error.

Happens in browser and in cockpit clients

**Comment by LionUCS** [comment]
I'll post the os-release since I'm piggybacking here:

```
NAME=EndeavourOS
PRETTY_NAME=EndeavourOS
ID=endeavouros
ID_LIKE=arch
BUILD_ID=2023.05.28
ANSI_COLOR="38;2;23;147;209"
HOME_URL='https://endeavouros.com'
DOCUMENTATION_URL='https://discovery.endeavouros.com'
SUPPORT_URL='https://forum.endeavouros.com'
BUG_REPORT_URL='https://forum.endeavouros.com/c/arch-based-related-questions/bug-reports'
PRIVACY_POLICY_URL=https://endeavouros.com/privacy-policy-2
LOGO=endeavouros
```

**Comment by martinpitt** [comment]
Originally I thought it would just stumble over a trailing empty line, but that's not it. I tested with your os-release, and can reproduce the bug. It's this line:

     PRIVACY_POLICY_URL=https://endeavouros.com/privacy-policy-2

When enclosing the value in single or double quotes (like all the other URLs), it works. So I have enough information to create a test and fix this, thanks!

**Comment by LionUCS** [comment]
Quoting the value does indeed work

Seems like an oversight on EOS, I can't find any other OS (including older versions of EOS) with any unquoted URLs

**Comment by martinpitt** [comment]
Ah, the [manpage](https://www.man7.org/linux/man-pages/man5/os-release.5.html) says:

> Variable assignment values must be enclosed in double or single quotes if they include spaces, semicolons or other special characters outside of A–Z, a–z, 0–9.

So indeed that should be fixed in EOS as well. I figure it's enough for the bridge to just ignore invalid values.

**Comment by martinpitt** [comment]
Fixed in #19100, thanks again for reporting!
